File: Scripts/results/generated_code_results/RN_gpt/Entertainment/musicAPP2/ExploreScreen.js
import React from 'react';
import {
  View,
  Text,
  Image,
  Button,
  StyleSheet,
  ScrollView,
  TouchableOpacity
} from 'react-native';

const ExploreScreen = ({ navigation }) => {
  return (
    <ScrollView style={styles.container}>
      <Text style={styles.time}>9:41</Text>
      <Text style={styles.header}>Explore</Text>

      <Section title="Recomanded for you">
        <AlbumCover />
        <AlbumCover />
        <AlbumCover />
      </Section>

      <Section title="Recent Played">
        <RecentPlayedItem />
        <RecentPlayedItem />
        <RecentPlayedItem />
      </Section>

      <Section title="New Released">
        <NewReleasedItem />
        <NewReleasedItem />
        <NewReleasedItem />
      </Section>

      <Section title="Popular Artist">
        <ArtistItem name="Justin Bieber" />
        <ArtistItem name="Michael Jackson" />
        <ArtistItem name="Sam Smith" />
      </Section>

      <View style={styles.genresSection}>
        <Text style={styles.sectionHeader}>Genres & Moods</Text>
        <GenreButton title="Hip Hop" />
        <GenreButton title="Pop" />
        <GenreButton title="Partyholics" />
      </View>

      <NowPlayingBar />
      <NavBar navigation={navigation} />
    </ScrollView>
  );
};

const Section = ({ title, children }) => (
  <View style={styles.section}>
    <Text style={styles.sectionHeader}>{title}</Text>
    <View style={styles.sectionContent}>{children}</View>
  </View>
);

const AlbumCover = () => (
  <View style={styles.albumCover}>
    <Image source={require('../assets/snack-icon.png')} style={styles.image} />
    <Text style={styles.albumTitle}>Baghi 3</Text>
    <Text style={styles.artist}>Various Artists</Text>
  </View>
);

const RecentPlayedItem = () => (
  <View style={styles.recentPlayed}>
    <Image source={require('../assets/snack-icon.png')} style={styles.image} />
    <Text style={styles.songTitle}>No Time To Die</Text>
  </View>
);

const NewReleasedItem = () => (
  <View style={styles.newReleased}>
    <Image source={require('../assets/snack-icon.png')} style={styles.image} />
    <Text style={styles.albumTitle}>Baghi 3</Text>
    <Text style={styles.artist}>Various Artists</Text>
  </View>
);

const ArtistItem = ({ name }) => (
  <View style={styles.artistItem}>
    <Image source={require('../assets/snack-icon.png')} style={styles.image} />
    <Text style={styles.artistName}>{name}</Text>
  </View>
);

const GenreButton = ({ title }) => (
  <TouchableOpacity style={styles.genreButton}>
    <Text style={styles.buttonText}>{title}</Text>
  </TouchableOpacity>
);

const NowPlayingBar = () => (
  <View style={styles.nowPlayingBar}>
    <Image source={require('../assets/snack-icon.png')} style={styles.nowPlayingImage} />
    <View style={styles.nowPlayingInfo}>
      <Text style={styles.nowPlayingTitle}>Illegal Weapon 2.0</Text>
      <Text style={styles.nowPlayingArtist}>Street Dancer 3D</Text>
    </View>
    <Button title="♥" onPress={() => {}} /> 
    <Button title="⏸️" onPress={() => {}} />
  </View>
);

const NavBar = ({ navigation }) => (
  <View style={styles.navBar}>
    <Button title="Home" onPress={() => navigation.navigate('Explore')} />
    <Button title="Tredning" onPress={() => navigation.navigate('Trending')} />
    <Button title="Search" onPress={() => navigation.navigate('Search')} />
    <Button title="Library" onPress={() => {}} />
    <Button title="Settings" onPress={() => {}} />
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 20,
  },
  time: {
    fontSize: 12,
    color: '#555',
  },
  header: {
    fontSize: 32,
    fontWeight: 'bold',
    marginBottom: 20,
  },
  section: {
    marginBottom: 20,
  },
  sectionHeader: {
    fontSize: 20,
    fontWeight: 'bold',
    marginBottom: 10,
  },
  sectionContent: {
    flexDirection: 'row',
  },
  albumCover: {
    marginRight: 10,
    width: 100,
  },
  image: {
    width: '100%',
    height: 100,
    borderRadius: 8,
  },
  albumTitle: {
    fontSize: 14,
    marginTop: 5,
  },
  artist: {
    fontSize: 12,
    color: '#666',
  },
  recentPlayed: {
    marginRight: 10,
    alignItems: 'center',
  },
  songTitle: {
    fontSize: 12,
    marginTop: 5,
  },
  newReleased: {
    marginRight: 10,
    alignItems: 'center',
  },
  artistItem: {
    marginRight: 10,
    alignItems: 'center',
  },
  artistName: {
    fontSize: 12,
    marginTop: 5,
  },
  genresSection: {
    marginBottom: 20,
  },
  genreButton: {
    backgroundColor: '#e0e0e0',
    padding: 10,
    borderRadius: 20,
    marginRight: 10,
  },
  buttonText: {
    fontSize: 14,
  },
  nowPlayingBar: {
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: '#e0e0e0',
    padding: 10,
    borderRadius: 10,
    marginVertical: 20,
  },
  nowPlayingInfo: {
    flex: 1,
    marginLeft: 10,
  },
  nowPlayingTitle: {
    fontSize: 16,
  },
  nowPlayingArtist: {
    fontSize: 14,
    color: '#666',
  },
  nowPlayingImage: {
    width: 50,
    height: 50,
    borderRadius: 25,
  },
  navBar: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    paddingVertical: 10,
  },
});

export default ExploreScreen;

File: Scripts/results/generated_code_results/Flutter/Entertainment/musicAPP2/ExploreScreen.dart
import 'package:flutter/material.dart';
class ExploreScreen extends StatefulWidget {
@override
_ExploreScreenState createState() => _ExploreScreenState();
}
class _ExploreScreenState extends State<ExploreScreen> {
@override
Widget build(BuildContext context) {
return Scaffold(
appBar: AppBar(
title: Text('9:41'),
actions: [
Icon(Icons.signal_cellular_alt),
SizedBox(width: 8),
Icon(Icons.wifi),
],
),
body: SingleChildScrollView(
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Padding(
padding: const EdgeInsets.all(16.0),
child: Text(
'Explore',
style: TextStyle(fontSize: 24, fontWeight: FontWeight.bold),
),
),
Padding(
padding: const EdgeInsets.symmetric(horizontal: 16.0),
child: Text(
'Recomanded for you',
style: TextStyle(fontSize: 18, fontWeight: FontWeight.w500),
),
),
SizedBox(height: 8),
SingleChildScrollView(
scrollDirection: Axis.horizontal,
child: Row(
children: [
_buildRecommendedItem('Baghi 3', 'Various Artists'),
_buildRecommendedItem('Surma Surma', 'Various Artists'),
_buildRecommendedItem('Hindi T', 'Various Artists'),
],
),
),
SizedBox(height: 16),
Padding(
padding: const EdgeInsets.symmetric(horizontal: 16.0),
child: Text(
'Recent Played',
style: TextStyle(fontSize: 18, fontWeight: FontWeight.w500),
),
),
SizedBox(height: 8),
SingleChildScrollView(
scrollDirection: Axis.horizontal,
child: Row(
children: [
_buildRecentPlayedItem('No Time To Die'),
_buildRecentPlayedItem('Illegal Weapon'),
_buildRecentPlayedItem('Loca'),
],
),
),
SizedBox(height: 16),
Padding(
padding: const EdgeInsets.symmetric(horizontal: 16.0),
child: Text(
'New Released',
style: TextStyle(fontSize: 18, fontWeight: FontWeight.w500),
),
),
SizedBox(height: 8),
SingleChildScrollView(
scrollDirection: Axis.horizontal,
child: Row(
children: [
_buildNewReleasedItem('Baghi 3', 'Various Artists'),
_buildNewReleasedItem('Baghi 3', 'Various Artists'),
],
),
),
SizedBox(height: 16),
Padding(
padding: const EdgeInsets.symmetric(horizontal: 16.0),
child: Text(
'Popular Artist',
style: TextStyle(fontSize: 18, fontWeight: FontWeight.w500),
),
),
SizedBox(height: 8),
SingleChildScrollView(
scrollDirection: Axis.horizontal,
child: Row(
children: [
_buildPopularArtistItem('Justin Bieber'),
_buildPopularArtistItem('Michael Jackson'),
_buildPopularArtistItem('Sam Smith'),
],
),
),
SizedBox(height: 16),
Padding(
padding: const EdgeInsets.symmetric(horizontal: 16.0),
child: Text(
'Genres & Moods',
style: TextStyle(fontSize: 18, fontWeight: FontWeight.w500),
),
),
SizedBox(height: 8),
SingleChildScrollView(
scrollDirection: Axis.horizontal,
child: Row(
children: [
_buildGenreMoodButton('Hip Hop'),
_buildGenreMoodButton('Pop'),
_buildGenreMoodButton('Partyholics'),
],
),
),
],
),
),
bottomNavigationBar: BottomNavigationBar(
currentIndex: 0,
items: [
BottomNavigationBarItem(icon: Icon(Icons.home), label: 'Home'),
BottomNavigationBarItem(
icon: Icon(Icons.trending_up),
label: 'Trending',
),
BottomNavigationBarItem(icon: Icon(Icons.search), label: 'Search'),
BottomNavigationBarItem(icon: Icon(Icons.library_music), label: 'Library'),
BottomNavigationBarItem(icon: Icon(Icons.settings), label: 'Settings'),
],
onTap: (index) {
if (index == 1) {
Navigator.pushNamed(context, 'Trending');
} else if (index == 2) {
Navigator.pushNamed(context, 'Search');
}
},
),
);
}
Widget _buildRecommendedItem(String title, String artist) {
return GestureDetector(
onTap: () => Navigator.pushNamed(context, 'NowPlaying'),
child: Container(
width: 150,
margin: EdgeInsets.only(left: 16),
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
ClipRRect(
borderRadius: BorderRadius.circular(8),
child: Image.asset('assets/images/template.png', height: 150, width: 150, fit: BoxFit.cover),
),
SizedBox(height: 4),
Text(title, style: TextStyle(fontWeight: FontWeight.bold)),
Text(artist, style: TextStyle(color: Colors.grey)),
],
),
),
);
}
Widget _buildRecentPlayedItem(String title) {
return GestureDetector(
onTap: () => Navigator.pushNamed(context, 'NowPlaying'),
child: Container(
width: 150,
margin: EdgeInsets.only(left: 16),
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
ClipRRect(
borderRadius: BorderRadius.circular(8),
child: Image.asset('assets/images/template.png', height: 150, width: 150, fit: BoxFit.cover),
),
SizedBox(height: 4),
Text(title, style: TextStyle(fontWeight: FontWeight.bold)),
],
),
),
);
}
Widget _buildNewReleasedItem(String title, String artist) {
return GestureDetector(
onTap: () => Navigator.pushNamed(context, 'NowPlaying'),
child: Container(
width: 150,
margin: EdgeInsets.only(left: 16),
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
ClipRRect(
borderRadius: BorderRadius.circular(8),
child: Image.asset('assets/images/template.png', height: 150, width: 150, fit: BoxFit.cover),
),
SizedBox(height: 4),
Text(title, style: TextStyle(fontWeight: FontWeight.bold)),
Text(artist, style: TextStyle(color: Colors.grey)),
],
),
),
);
}
Widget _buildPopularArtistItem(String name) {
return Container(
width: 100,
margin: EdgeInsets.only(left: 16),
child: Column(
children: [
ClipOval(
child: Image.asset('assets/images/template.png', height: 100, width: 100, fit: BoxFit.cover),
),
SizedBox(height: 4),
Text(name, style: TextStyle(fontWeight: FontWeight.bold), textAlign: TextAlign.center),
],
),
);
}
Widget _buildGenreMoodButton(String label) {
return Container(
margin: EdgeInsets.only(left: 16),
child: ElevatedButton(
onPressed: () {},
child: Text(label),
style: ElevatedButton.styleFrom(
backgroundColor: Colors.pink,
shape: RoundedRectangleBorder(
borderRadius: BorderRadius.circular(20),
),
),
),
);
}
}